contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -1203,15 +1205,43 @@ is_numeric_datatype(Oid typid)
 	return false;
 }
 
+/*
+ * get_default_typmod_for_fixedsize_dataypes
+ *
+ * Assigns predefined typmod values for fixed-length datatypes
+ * (int, bigint, smallint, tinyint) and money/smallmoney.
+ * These typmods represent the maximum allowed precision for each type
+ * and are used during typmod resolution to ensure consistent handling
+ * in expressions involving fixed-size numeric types.
+ */
+static int32
+get_default_typmod_for_fixedsize_dataypes(Oid resulttype)
+{
+	if ((*common_utility_plugin_ptr->is_tsql_money_datatype)(resulttype))
+		return TSQL_MONEY_TYPMOD;
+	else if ((*common_utility_plugin_ptr->is_tsql_smallmoney_datatype)(resulttype))
+		return TSQL_SMALLMONEY_TYPMOD;
```

Comment:
For efficiency, we should have most common and easiest checks at the top, i.e. INTOIDs and then Money at the bottom where we might be looking up